### PR TITLE
Dashboard

### LINKED ISSUE #10: Dashboard: Viewing Parties
As an authenticated user,
I should see the viewing parties I have been **invited** to with the following details:
- Movie Title
- Date and Time of Event
- Status of Invited

I should also see the viewing parties that I have **created** with the following details:
- Movie Title
- Date and Time of Event
- Status of Host

### LINKED ISSUE #11: Dashboard: Friends
As an authenticated user,
I see a section for friends,
In this section, there should be a text field to enter a friend's email and a button to "Add Friend"

**Scenarios**: 
1. If I have not added any friends there should be a message. "You currently have no friends".

2. If I have added friends, I should see a list of all my friends.

**Details:** Users should be able to add a friend by their email address, as long as, the friend is a user of our application and exists in our database. 

**Example:** 
Bugs Bunny and Lola Bunny are users of our application, but Daffy Duck is not. 
1. When Bugs Bunny enters `lola_bunny@gmail.com` to add friend it will be successful and Lola should show up as Bugs Bunny's friend.
2. When Bugs Bunny enters `daffy_duck@gmail.com` to add friend it should give an error message that the user does not exist. 

**Tips:** You'll want to research self-referential `has_many through`. [Here](http://blog.hasmanythrough.com/2007/10/30/self-referential-has-many-through) is a good starting point to understand the concept. You will probably need to do more googling but that's part of the fun ;)

- [x] Write a happy path test 
- [x] Write a sad path test

### LINKED ISSUE #12: Dashboard: Discover Movies
As a authenticated user, 
When I click on the "Discover Movies" button,
I am redirected to a discover page.

### LINKED ISSUE #13: Dashboard
As an authenticated user,
When I visit '/dashboard'
I should see: 
- [x] 'Welcome `<username>`!' at the top of page
- [x] #10 A friends section
- [x] #9  A button to Discover Movies
- [x] #11 A viewing parties section

### PR DESCRIPTION
**Pull request by smcunning**
Creates Migrations - 
- Friendships
- Parties 
- Guests

Views - 
- Creates Dashboard view with the following functionality: 
    - Search by email for users in the database to add as friends 
    - Sad paths tested for adding users who don't exist, adding yourself as a friend and adding multiples of the same user 
    - Displays all parties and basic details for every party user is associated with, determining whether Host or Invited guest 

Testing - 
- All tests currently passing 

closes #13 
closes #11 
closes #10
closes #12

### CODE REVIEW

File: app/controllers/dashboard_controller.rb [hunk 1]
```diff
@@ -1,10 +1,17 @@
 class DashboardController < ApplicationController
   before_action :require_user
+
   def index
+    # require "pry"; binding.pry
```
Comment (KieraAllen) [inline app/controllers/dashboard_controller.rb]: remove

File: app/controllers/dashboard_controller.rb [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+    guests = @current_user.guests
+    @parties = guests.map do |guest|
+      Party.find_by(id: guest.party_id)
+    end
+    # require "pry"; binding.pry
```
Comment (KieraAllen) [inline app/controllers/dashboard_controller.rb]: remove

File: app/controllers/dashboard_controller.rb (hunk 1 above)
Comment (araspaas) [inline app/controllers/dashboard_controller.rb]: We can remove these comments

File: app/controllers/friendships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class FriendshipsController < ApplicationController
+
+  def create
+    new_friend = User.find_by(email: params[:email])
+
+    create_friendship(current_user.id, new_friend)
+    redirect_to "/dashboard"
+  end
+
+  def create_friendship(user_id, friend)
```
Comment (KieraAllen) [inline app/controllers/friendships_controller.rb:10]: Would this (and `friend_exists?`) work as private methods?

Comment (araspaas) [inline app/controllers/friendships_controller.rb:10]: we can Stick this in a private section

**Review by araspaas — APPROVED** [review]

**Pull request merged**